hbase-client/src/main/java/org/apache/omid/transaction/TTable.java

```diff
@@ -396,20 +443,48 @@ public ResultScanner getScanner(Transaction tx, Scan scan) throws IOException {
         return commitCache;
     }
 
-    private boolean isCellInSnapshot(Cell kv, HBaseTransaction transaction, Map<Long, Long> commitCache)
-        throws IOException {
+    private void buildFamilyDeletionCache(List<Cell> rawCells, Map<String, List<Cell>> familyDeletionCache) {
 
-        long startTimestamp = transaction.getStartTimestamp();
+        for (Cell cell : rawCells) {
+            if (CellUtil.matchingQualifier(cell, CellUtils.FAMILY_DELETE_QUALIFIER) &&
+                    CellUtil.matchingValue(cell, HConstants.EMPTY_BYTE_ARRAY)) {
+
+                String row = Bytes.toString(cell.getRow());
+                List<Cell> cells = familyDeletionCache.get(row);
```

Comment:
Please decide if you return null or Optional from methods, the code looks non-uniform